src/pages/iou/request/step/IOURequestStepTag.js

```diff
@@ -80,9 +82,9 @@ function IOURequestStepTag({
      */
     const updateTag = (selectedTag) => {
         const isSelectedTag = selectedTag.searchText === tag;
-        const updatedTag = !isSelectedTag ? selectedTag.searchText : '';
+        const updatedTag = IOUUtils.insertTagIntoReportTagsString(transactionTag, isSelectedTag ? '' : selectedTag.searchText, tagIndex);
```

Comment:
Missed case: If we are using a multi level tag and then switched to a single level tag. The current IOU would still have the multi level tags with a violation e.g. `tag1:tag2` (which is fine). Now if the user changes the first tag to fix the violation, the new tags should contain just `newTag` but the current logic does not take that case into account and just replaces the first tag i.e. `newTag,tag2` which still yield a violation that can't be fixed.

Coming from https://github.com/Expensify/App/issues/68263